### ISSUE TITLE
Colony-level variances to individual-level variances

### ISSUE THREAD

**Issue by gregorgorjanc** [HighlanderLab/SIMplyBee#49]
Check how ignoring covariances between workers is impacting transformation of the variance between average worker effects Var(avg(a_w)) (German estimate of 2 for some trait in some data) to variance between individual worker effects Var(a_w)! A simple transformation is Var(a_w)=n*Var(avg(a_w)) , but this is an overestimate that assumes the workers are not related, which is not true - they are either sisters or supersisters. A simple way would be to assume a proportion of sisters (cov=0.5) and super-sisters (cov=0.75), then get average cov and update Var(a_w)=n*Var(avg(a_w))

**Comment by gregorgorjanc** [comment]
As mentioned before, I think there is lots of material here that I think we could publish a paper on this. I do however wonder how should we pitch such a paper. Options:

1) Standard short note paper: Here is the problem, here is the solution, here is a Shiny app that implements the solution

2) 1) but with a supplementary, which shows how we derived the solution (contents of the current messy! vignette, obviously polished) - the idea here is that we show how useful SIMplyBee can be for the development of theory

3) Demonstration of how useful SIMplyBee can be for the development of theory as the main paper? This will be hard for the reviewers to accept and maybe for readers to appreciate, though I am actually thinking that some students and many researchers would appreciate such an approach more than the "here is the solution" paper (1 or 2 above).

Likely we will struggle to get 3) published, so maybe best if we opt for 2. Any thoughts @janaobsteter ?

**Comment by janaobsteter** [comment]
Yes, I do agree with 2! 3 could be problematic since we simulate what we think it's true and then we want to confirm our theory with the simulation - it's sort of chicken and egg situation? But 2 should be great! I think ShinyApp would make a great addition :)

**Comment by gregorgorjanc** [comment]
OK, 2 it is.

Just to add, that we do want theory and simulation to match since they are based on the same underlying genetic model - that the genetic or phenotypic value of a colony is a sum of queen and worker genetic or phenotypic values.

**Comment by janaobsteter** [comment]
Noted from the meeting:

1. Going from individual to colony variance
We have 6 parameters: 
- three genetic: $\sigma_{g_q}^2$ for queen effect,  $\sigma_{g_w}^2$ for worker effect, and the covariance $\sigma_{g_q,g_w}$
- three environmental:  $\sigma_{e_q}^2$ for queen effect,  $\sigma_{e_w}^2$ for worker effect, and the covariance $\sigma_{e_q,e_w}$
- for transforming these into colony level values we would also potentially need either number of workers ($n_w$) number of fathers ($n_f$) & number of DPQs ($n_{DPQ}$), or average relatedness between workers in the colony (as in BB)

2. Going from colony to individual variance
- here, it depends what the papers report
- option1: they only report one variance, which would be $\sigma_{g}^2$, that would be reported as the queen trait - but estimated from colony phenotype, and  $\sigma_{E}^2$
- option2: at best, they would report $\sigma_{g_q}^2$, $\sigma_{g_w}^2$, $\sigma_{g_q, g_w}$, and $\sigma_{E_c}^2$. This makes it four parameters, but we need 6 to simulate individual variances. How to then tear apart the environmental variance? Options:
  -  each of the $\sigma_{E_q}^2$,  $\sigma_{E_q}^2$, and  $\sigma_{E_q, E_w}$ is 1/3 of  $\sigma_{E_c}^2$
  - each of the $\sigma_{E_q}^2$ and $\sigma_{E_q}^2$ are 1/2 of $\sigma_{E_c}^2$, and  $\sigma_{E_q, E_w}$ is 0
  - the user defines p1, p2, and p3 that represent the proportion of $\sigma_{E_c}^2$ assigned to $\sigma_{E_q}^2$,  $\sigma_{E_q}^2$, and  $\sigma_{E_q, E_w}$

<img width="993" height="1319" alt="Image" src="https://github.com/user-attachments/assets/20a102e9-c637-4305-bba4-443a57eb550d" />

**Comment by janaobsteter** [comment]
Suggestions for the paper title: "A note on individual and colony/group quantitative genetic analysis in honeybees/social organisms/eusocial species"

**Comment by gregorgorjanc** [comment]
![Image](https://github.com/user-attachments/assets/25c9fe50-fd8e-4eb1-bcf8-4d8a0cad6ae9)

**Comment by janaobsteter** [comment]
Report from the hackathon in Slovenia in May, 2026. 

We created functions:
-  `mapIndToColonyVariance` that takes varA/varE for the queen, varA/varE for individual workers, corA/covE between them and calculates the expected colony level variances
-  `mapColonyToindVariance` that takes varA/varE for the queen, varA/varE for the worker group (wbar), and corA/covE between them and calculates the components that need to be set for simulating individual workers in SIMplyBee to results in the expected colony-level values.

The logic is explained in F2_Variance_calculations.Rmd, F2_Variance_calculations_functions.Rmd.
We tested both functions:
-  `mapIndToColonyVariance` by setting the individual components, calculating the expected colony components, then simulating the population and collecting the realised components when the aggregate function for worker group effect is either `mean` or a `sum`

<img width="1187" height="734" alt="Image" src="https://github.com/user-attachments/assets/1058cd6f-f337-4b33-8dfd-1f876e312a4f" />

-  `mapColonyToindVariance` by setting expected colony components, then calculating individual components, simulating the population with those components and collecting the realised components when the aggregate function for worker group effect is either `mean` or a `sum`

<img width="1187" height="734" alt="Image" src="https://github.com/user-attachments/assets/840b7075-51b4-4bda-bdee-8f766886aacb" />

Further work is described in #636
